extension/lib/shopify/collection-repository/command/ListGraphQl.js

```diff
@@ -0,0 +1,70 @@
+// const query = require('./Get-GraphQL-query')
+
+class ShopifyCollectionRepositoryCommandListGraphQL {
+  /**
+   * @param {ShopifyStorefrontClient} client
+   */
+  constructor (client) {
+    this._storefrontClient = client
+  }
+
+  /**
+   *
+   * @param {string} id
+   * @returns {Promise<ShopifyCollectionRepositoryCommandGetOutput[]>}
+   */
+  async execute () {
+    console.log('doing execute in listGraphQl')
+    const query = this._storefrontClient.query((root) => {
+      root.add('shop', shop => {
+        shop.add('collections', {
+          args: {
+            first: 250
+          }
+        }, collections => {
+          collections.add('edges', edges => {
+            edges.add('node', node => {
+              node.add('id')
+              node.add('title')
+              node.add('handle')
+              node.add('image', image => {
+                image.add('originalSrc')
+                image.add('transformedSrc')
+              })
+            })
+          })
+          collections.add('pageInfo', pageInfo => {
+            pageInfo.add('hasNextPage')
+            pageInfo.add('hasPreviousPage')
+          })
+        })
+      })
+    })
+
+    try {
```

Comment:
Unless we have a logic to rewrite the error or handle it there should be no try/catch here.

Reply:
done